sequence/fasta_io.hpp

```diff
@@ -6,25 +6,39 @@
 #include <filesystem>
 #include <fstream>
 #include <iostream>
-#include <string>
+#include <iterator>
 #include <sstream>
+#include <string>
 
 namespace ts { // ts = Tensor Sketch
 
+// Represents the contents of a single Fasta file.
+// All the sequences in a file should be treated as a single assembly and should be sketched as a
+// whole.
+template <typename seq_type>
+struct FastaFile {
+    // The name of the file
+    std::string filename;
+    // The leading comment before each sequence. Always has the same length as sequences.
+    std::vector<std::string> comments;
+    // The sequences in the file
+    std::vector<std::vector<seq_type>> sequences;
+};
+
 /**
- * Reads a fasta file and returns its contents as a tuple of <test_id, sequences, names>.
+ * Reads a fasta file and returns its contents.
  * @tparam seq_type type used for storing a character of the fasta file, typically uint8_t
  */
 template <typename seq_type>
-std::pair<Vec2D<seq_type>, std::vector<std::string>> read_fasta(const std::string &file_name,
-                                                                const std::string &input_format) {
+FastaFile<seq_type> read_fasta(const std::string &file_name, const std::string &input_format) {
+    FastaFile<seq_type> f;
+    f.filename = std::filesystem::path(file_name).filename();
```

Comment:
Maybe move this block after the next `if` statement, as there is no reason to compute the filename if we later exit.